### PR TITLE
Additional sort-limit tests

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

These are strictly additional tests for the constrained heap sort in AQL. Motivated by https://github.com/arangodb/arangodb/pull/9981 mainly to have some more tests for sort-limit with fullCount.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6246/

File: arangod/Aql/ConstrainedSortExecutor.cpp [hunk 1]
```diff
@@ -209,7 +209,9 @@ std::pair<ExecutionState, NoStats> ConstrainedSortExecutor::produceRows(OutputAq
       return {ExecutionState::DONE, NoStats{}};
     }
     // We should never get here, as the following LIMIT block should never fetch
-    // more than our limit. He may only skip after that. Thus:
+    // more than our limit. He may only skip after that.
```
Comment (markuspf) [inline arangod/Aql/ConstrainedSortExecutor.cpp]: Nitpick: in english the block is neuter.
```suggestion
    // more than our limit. It may only skip after that.
```

File: tests/js/server/aql/aql-profiler.js [hunk 2]
```diff
@@ -81,40 +83,84 @@ function ahuacatlProfilerTestSuite () {
...
   const additionalLimitTestRowCounts = [
     // limit() = 1000 ± 1:
     1332, 1333, 1334,
-    // skip() = 1000 ± 1:
+    // offset() = 1000 ± 1:
     3999, 4000, 4003, 4004,
     // limit() = 2000 ± 1:
     2665, 2666, 2667,
-    // skip() = 2000 ± 1:
+    // offset() = 2000 ± 1:
```
Comment (markuspf) [inline tests/js/server/aql/aql-profiler.js]:
```suggestion
    // offset() = 8000 ± 1:
```

Comment (markuspf) [reply tests/js/server/aql/aql-profiler.js]: ?

File: tests/js/server/aql/aql-profiler.js [hunk 3]
```diff
@@ -81,40 +83,84 @@ function ahuacatlProfilerTestSuite () {
     UpsertBlock, ScatterBlock, DistributeBlock, IResearchViewUnorderedBlock,
     IResearchViewBlock, IResearchViewOrderedBlock } = profHelper;
 
-  // See the limit tests (e.g. testLimitBlock3) for limit() and skip().
+  // See the limit tests (e.g. testLimitBlock3) for limit() and offset().
   const additionalLimitTestRowCounts = [
     // limit() = 1000 ± 1:
     1332, 1333, 1334,
-    // skip() = 1000 ± 1:
+    // offset() = 1000 ± 1:
```
Comment (markuspf) [inline tests/js/server/aql/aql-profiler.js]:
```suggestion
    // offset() = 4000 ± 1:
```

Comment (markuspf) [reply tests/js/server/aql/aql-profiler.js]: ?

Comment (goedderz) [reply tests/js/server/aql/aql-profiler.js]: What's meant is, that for `rows` in `[3999, 4000, 4003, 4004]`, `offset(rows) = 1000 ± 1`.

File: tests/js/server/aql/aql-profiler.js (hunk 2 above)
Comment (goedderz) [reply tests/js/server/aql/aql-profiler.js]: See above.

File: tests/js/server/aql/aql-profiler.js (hunk 3 above)
Comment (goedderz) [reply tests/js/server/aql/aql-profiler.js]: The idea is to choose `rows` in a way, that `offset(rows)` is just around the size of a batch, so we catch those corner cases.

**Comment by goedderz** [comment]
Jenkins is green.
